tests/conftest.py

```diff
@@ -307,10 +307,21 @@ def finalizer():
         """
         Delete the PVC
         """
+        pv_objs = []
+
+        # Get PV form PVC instances and delete PVCs
         for instance in instances:
-            instance.delete()
-            instance.ocp.wait_for_delete(
-                instance.name
+            if not instance.is_deleted:
+                pv_objs.append(instance.backed_pv_obj)
+                instance.delete()
+                instance.ocp.wait_for_delete(
+                    instance.name
+                )
+
+        # Wait for PVs to delete
+        for pv_obj in pv_objs:
+            pv_obj.ocp.wait_for_delete(
```

Comment:
If the reclaim policy is retain it will fail

Reply:
I think we can expect that in this teardown phase, all PVs should be deleted, even if the PVCs reclaim policy is retain (by the test itself or with a teardwon that should take place before this teardown).

Reply:
not sure I get it.
if this is the expectation that all PVs will be deleted by now (this point in code), why do we need to add this wait_for_delete?

Reply:
but in the case of 'delete' reclaim policy, in the tests, we don't explicitly delete the PVs.
Also, the main reason for this PR is https://bugzilla.redhat.com/show_bug.cgi?id=1726266

Reply:
@jilju 
You can check the reclaim policy the backed_reclaim_policy added here https://github.com/red-hat-storage/ocs-ci/pull/584/files#diff-31d76ef98c9184ea833bdd9302eba288R104
and validate PV deletion only in case the policy is 'delete'

Reply:
Or PV should be explicitly deleted here ?

Reply:
If reclaim policy is retain and PV is not deleted, then  teardown will fail in leftover check.

Reply:
@jilju I agree. In case of retain policy, PV deletion should be taken care of in the test or in a teardwon that takes place before pvc_factory

Reply:
ack.
We will need to have issue on enhancing the log in ocp.wait_for_delete() to describe the object when fail to delete and also add the reclaim policy after @shreekarSS PR is in.
Lste's merge this one